src/transformers/models/bark/modeling_bark.py

```diff
@@ -0,0 +1,1372 @@
+
+        x_semantic_history = torch.from_numpy(x_semantic_history).to(self.device)
+        x_coarse_history = torch.from_numpy(x_coarse_history).to(self.device)
+
+        return x_semantic_history, x_coarse_history
+
+    def generate_text_semantic(
+        self,
+        inputs: torch.Tensor,
+        history_prompt: Optional[Dict[str, np.ndarray]] = None,
+        **kwargs,
+    ) -> torch.LongTensor:
+        # TODO: add a max_gen_duration_s early stop
+
+        input_ids = inputs
+
+        # TODO: Not used for now. where to set the default value ?
+        # min_eos_p = kwargs.get("min_eos_p", 0.2)
+
+        # TODO: input_ids[:,256:256+256] (to verify) corresponds to history_prompt["semantic_prompt"], maybe use that
+        # input_ids should be of shape (batch_size, seq_len) where seq_len = 513
+        input_embeds = torch.cat(
+            [
+                self.semantic_model.transformer.wte(input_ids[:, :256])
+                + self.semantic_model.transformer.wte(input_ids[:, 256 : 256 + 256]),
+                self.semantic_model.transformer.wte(input_ids[:, 256 + 256 :]),
+            ],
+            dim=1,
+        )
+
+        semantic_logits_processor = SemanticLogitsProcessor(
+            self.config.semantic_vocab_size, self.config.semantic_pad_token
+        )
+
+        # TODO: for now, it is not implemented yet as long as StoppingCriteria issue is not dealt with
+        # https://github.com/huggingface/transformers/issues/23674
+        # semantic_stopping_criteria = SemanticStoppingCriteria(min_eos_p , self.config.semantic_pad_token)
+
+        # pass input_ids in order to stay consistent with the transformers generate method even though it is not used (except to get the input seq_len - that's why we keep the first 257 tokens)
+        semantic_output = self.semantic_model.generate(
+            input_ids[:, :257],
+            input_embeds=input_embeds,
+            logits_processor=[semantic_logits_processor],
+            # stopping_criteria=[semantic_stopping_criteria],
+            renormalize_logits=True,
+            eos_token_id=self.config.semantic_pad_token,
+            max_new_tokens=768,
+            output_scores=False,
+            return_dict_in_generate=False,
+            output_hidden_states=False,
+            output_attentions=False,
+            **kwargs,
+        )  # size: 10048
+        # TODO: there is also a max_gen_duration_s early stop if the duration depass a certain duration
+
+        # take the generated semantic tokens
+        semantic_output = semantic_output[:, 257:]
+        
+        # to stay consistent with Bark original library, replace semantic_pad_token (eos_tok and pad_tok here) with coarse_semantic_pad_token i.e the pad_token used in the next model
+        semantic_output.masked_fill_(semantic_output == self.config.semantic_pad_token, self.config.coarse_semantic_pad_token)
+
+        return semantic_output
+
+    def generate_coarse(
```

Comment:
Likewise, can be tied to the coarse model

Reply:
What do you mean?

Reply:
IMO it makes sense for this method to belong to the coarse model, not the composite one. If we loaded the coarse model standalone, how would we generate currently? If we change to putting this method under the coarse model, then we can load the coarse model standalone and generate with it.